includes/woopay/services/class-checkout-service.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Class Create_And_Confirm_Intention_Test
+ *
+ * @package Checkout_Service
+ */
+
+namespace WCPay\WooPay\Service;
+
+use WC_Payments_Features;
+use WCPay\Core\Exceptions\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WCPay\Core\Server\Request\WooPay_Create_And_Confirm_Intention;
+
+/**
+ * Checkout service class.
+ */
+class Checkout_Service {
+
+	/**
+	 * Create woopay request from base create and confirm request.
+	 *
+	 * @param Request   $base_request Base request.
+	 * @param \WC_Order $order Order.
+	 * @param bool      $using_saved_payment_method Using saved payment method.
+	 *
+	 * @return WooPay_Create_And_Confirm_Intention
+	 * @throws Invalid_Request_Parameter_Exception
+	 * @throws \WCPay\Core\Exceptions\Extend_Request_Exception
+	 */
+	public function create_woopay_intention_request( $base_request, $order, $using_saved_payment_method ) {
+		if ( ! $order ) {
+			throw new Invalid_Request_Parameter_Exception(
+				'Invalid order passed',
+				'wcpay_core_invalid_request_parameter_order'
+			);
+		}
+		$request = WooPay_Create_And_Confirm_Intention::extend( $base_request );
+		$request->set_has_woopay_subscription( '1' === $order->get_meta( '_woopay_has_subscription' ) );
+		$request->set_is_platform_payment_method( self::is_platform_payment_method( $using_saved_payment_method ) );
+		return $request;
+	}
+
+	/**
+	 * Determine if current payment method is a platform payment method.
+	 *
+	 * @param boolean $is_using_saved_payment_method If it is using saved payment method.
+	 *
+	 * @return boolean True if it is a platform payment method.
+	 */
+	public static function is_platform_payment_method( bool $is_using_saved_payment_method ) {
```

Comment:
There is no point of having this static, let's instantiate the service.

Reply:
Fixed. These methods are not static anymore.